mathics/core/parser/tokeniser.py

```diff
@@ -230,7 +233,7 @@
     '<': ['Get', 'StringJoin', 'LessEqual', 'Less'],
     '=': ['SameQ', 'UnsameQ', 'Equal', 'Unset', 'Set'],
     '>': ['PutAppend', 'Put', 'GreaterEqual', 'Greater'],
-    '?': ['PatternTest'],
+    '?': ['Information', 'Definition','PatternTest'],
```

Comment:
So there are two tokens `('SingleQM',  ' \? ')` and `('DoubleQM', '  \?\? ')`. This part of the tokeniser looks at a single character and tries to determine what token will match.

Let's say the tokeniser sees `? next. The problem is we don't know which token to emit without looking at the next character. The next character is either `?` or it's something else. The solution is to first try matching `DoubleQM` and then try `SingleQM`.

Does that make sense?